discord/ext/alternatives/inline_bot_commands.py

```diff
@@ -0,0 +1,78 @@
+"""An experiment that allows you to define commands in the bot subclass
+
+Example:
+```py
+from discord.ext.alternatives import inline_bot_commands
+from discord.ext import commands
+
+class MyBot(commands.Bot):
+    @commands.command()
+    async def test(self, ctx):
+        await ctx.send(f'{len(self.all_commands)} commands registered on {self.__class__.__name__}!')
+
+    @commands.command()
+    async def echo(self, ctx, *, words):
+        await ctx.send(words)
+
+
+bot = MyBot(command_prefix='?')
+bot.run(token)
+```
+"""
+
+from discord.ext import commands
+
+
+class InlineMeta(type):
+    def __new__(cls, *args, **kwargs):
+        new_cls = super().__new__(cls, *args)
+        cmds = {}
+        for base in reversed(new_cls.__mro__):
+            for elem, value in base.__dict__.items():
+                if elem in cmds:
+                    del cmds[elem]
```

Comment:
Isn't there no point deleting the old reference as you replace it below?

Reply:
No. Consider the following:

```py
class Foo(commands.Bot):
    @commands.command()
    async def foo(self, ctx):
        ...

class Bar(Foo):
    async def foo(self, ctx):
        ...
```

Bar.foo overrides the old callback, but isn't a command object. The overriden callback is still registered as a command, so we must delete it.

Reply:
Oh right, that makes sense. 👍